public/app/features/explore/state/query.ts

```diff
@@ -222,7 +222,18 @@ export const clearCacheAction = createAction<ClearCachePayload>('explore/clearCa
 export function addQueryRow(exploreId: ExploreId, index: number): ThunkResult<void> {
   return async (dispatch, getState) => {
     const queries = getState().explore[exploreId]!.queries;
-    const query = await generateEmptyQuery(queries, index);
+    let datasourceOverride = undefined;
+
+    // if this is the first query being added, check for a root datasource
+    // if it's not mixed, send it as an override. generateEmptyQuery doesn't have access to state
+    if (queries.length === 0) {
+      const rootDatasource = getState().explore[exploreId]!.datasourceInstance;
+      if (!config.featureToggles.exploreMixedDatasource || !rootDatasource?.meta.mixed) {
```

Comment:
Nit: Maybe just a personal preference but I think the other way around reads a bit better:

```suggestion
      if (!(config.featureToggles.exploreMixedDatasource && rootDatasource?.meta.mixed)) {
```

also data source can be `mixed` only if the toggle is enabled so maybe we also just use:

```suggestion
      if (!rootDatasource?.meta.mixed) {
```